src/preprocess_data.py

```diff
@@ -0,0 +1,117 @@
+def pre_process(path):
+
+#Remove symbols and convert to nums for Wage,Value and Release Clause
+    fifa = pd.read_csv(path+'EA_FIFA19.csv')
+
+    def value_to_int(df_value):
+        try:
+            value = float(df_value[1:-1])
+            suffix = df_value[-1:]
+
+            if suffix == 'M':
+                value = value * 1000000
+            elif suffix == 'K':
+                value = value * 1000
+        except ValueError:
+            value = 0
+        return value
+
+    fifa['Value'] = fifa['Value'].apply(value_to_int)
+    fifa['Wage'] = fifa['Wage'].apply(value_to_int)
+
+    fifa['Release Clause'] = fifa['Release Clause'].fillna('0')
+    fifa['Release Clause'] = fifa['Release Clause'].apply(value_to_int)
+
+    def check_contract(row):
+        month_list=['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
+        ref_date = datetime(2018,5,31,0,0,0)
+        contract = row['Contract Valid Until']
+        try:
+            match = re.findall('(\w{3}) \d{1,2}, (\d{4})',contract)
```

Comment:
Don't you have to have imports in this code? where is import re?

Reply:
K